src/cdk/generator.ts

```diff
@@ -132,8 +132,10 @@ export class ResourceGenerator extends Construct {
     resources.forEach((resource) => this.generateConstructsForResource(resource))
 
     // it's handy to have the API base URL as a stack output
-    if (this.httpApi?.url)
-      new CfnOutput(this, "ApiBase", { value: this.httpApi.url, exportName: Fn.join("-", [Aws.STACK_NAME, "ApiBase"]) })
+    if (this.httpApi?.url) {
+      const apiName: string = this.httpApi.httpApiName || "ApiBase"
+      new CfnOutput(this, apiName, { value: this.httpApi.url, exportName: Fn.join("-", [Aws.STACK_NAME, apiName]) })
+    }
```

Comment:
```suggestion
      const apiName = this.httpApi.httpApiName
      new CfnOutput(this, apiName, { value: this.httpApi.url, ...(apiName ? {exportName: Fn.join("-", [Aws.STACK_NAME, apiName]) : {}) })
    }
```

Reply:
I need apiName to be `this.httpApi.httpApiName || "ApiBase"` because `this.httpApi.httpApiName` is type `string | undefined` and `CfnOutput` takes a string not undefined

Reply:
@revmischa I had to touch this line as I was getting a syntax error, also I added back the `|| "ApiBase"` let me know if there is another way for CfnOutput to take a `string | undefined` type or if there is a way to cast `string | undefined` to `string`